### PR TITLE
Readdition of UnlockingGeneratorEventArgs

### PR DESCRIPTION
**Pull request by BuildBoy12**

### CODE REVIEW

File: Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs [hunk 1]
```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
                 {
                     case PlayerInteract.Generator079Operations.Door:
                         bool isAllowed = true;
+                        if (!__instance.isDoorUnlocked)
+                        {
+                            isAllowed = person.GetComponent<ServerRoles>().BypassMode;
```
Comment (galaxy119) [inline Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs:39]: You wouldn't want to use a GetComponent<> here, simply use Player.Get(person).Bypass or Player.Get(person).ReferenceHub.serverRoles

File: Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs [hunk 2]
```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
...
+                            
+                            if (person.GetComponent<Inventory>().curItem > 0)
```
Comment (galaxy119) [inline Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs:41]: Again, no need for a GetComponent, they are expensive and unnecessary.

File: Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs [hunk 3]
```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
...
+                            {
+                                foreach (string permission in person.GetComponent<Inventory>().GetItemByID(person.GetComponent<Inventory>().curItem).permissions)
```
Comment (galaxy119) [inline Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs:43]: Same as above.

File: Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs [hunk 4]
```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
...
+                                {
+                                    if (permission == "ARMORY_LVL_2")
+                                        isAllowed = true;
+                                }
+                            }
+
+                            var unlockingEventArgs = new UnlockingGeneratorEventArgs(API.Features.Player.Get(person), __instance, isAllowed);
```
Comment (galaxy119) [inline Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs:50]: You never check to see if the generator is locked, therefore you will call this event everytime a closed generator door is open, regardless of if it's actually being unlocked.

File: Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs [hunk 5]
```diff
@@ -34,6 +34,38 @@ private static bool Prefix(Generator079 __instance, GameObject person, PlayerInt
...
+                            
+                            Player.OnUnlockingGenerator(unlockingEventArgs);
+                            
+                            isAllowed = unlockingEventArgs.IsAllowed;
+
+                            if (isAllowed)
```
Comment (galaxy119) [inline Exiled.Events/Patches/Events/Player/InsertingGeneratorTablet.cs:56]: This would simply unlock but never actually open the generator door.

**Comment by BuildBoy12** [comment]
In this last commit, I sincerely beg to differ about those errors. It'd throw errors if it the whitespace was there or not due to the rule on closing brackets.

**Comment by galaxy119** [comment]
image
It's referring to this whitespace.
The empty line is fine, but it should not contain a whitespace.

**Comment by BuildBoy12** [comment]
Noted, thank you much for the tips.

**Comment by BuildBoy12** [comment]
Closing due to manual additions.

**Comment by galaxy119** [comment]
Manually added in 7995a4c because github was being stupid.

**Pull request closed without merging**